### ISSUE TITLE
Broken boost::sort::spreadsort::string_sort

### ISSUE THREAD

**Issue by sgn** [boostorg/sort#66]
Below function
https://github.com/boostorg/sort/blob/db798f62caea5b6e56c98a12daeaf336e4e78561/include/boost/sort/spreadsort/detail/string_sort.hpp#L696-L705

Will be disable if `sizeof(Unsigned_char_type) <= 2`, that means that function will be enabled when `sizeof(Unsigned_char_type) > 2`.

However, when `sizeof(Unsigned_char_type) > 2`, the `BOOST_STATIC_ASSERT( sizeof(Unsigned_char_type) <= 2 );` will run into failure.

**Comment by spreadsort** [comment]
Good point, I missed that in the code review for
https://github.com/boostorg/sort/pull/40.  I'll remove the unnecessary
ASSERTs.

**Comment by spreadsort** [comment]
Thanks for spotting that!  The fix is in develop: https://github.com/boostorg/sort/commit/2e6674d63f2f541a3fd32f50d593be0c71f686f8

**Comment by spreadsort** [comment]
merge into master is here: https://github.com/boostorg/sort/pull/67, waiting on Francisco's review.
